scipy/stats/stats.py

```diff
@@ -3454,19 +3473,166 @@ def ttest_ind(a, b, axis=0, equal_var=True):
     if a.size == 0 or b.size == 0:
         return (np.nan, np.nan)
 
-    v1 = np.var(a, axis, ddof=1)
-    v2 = np.var(b, axis, ddof=1)
-    n1 = a.shape[axis]
-    n2 = b.shape[axis]
-
-    if equal_var:
-        df, denom = _equal_var_ttest_denom(v1, n1, v2, n2)
+    if random_state is None:
+        random_state = np.random.RandomState()
+    elif isinstance(random_state, int):
+        random_state = np.random.RandomState(seed=random_state)
```

Comment:
You can use `check_random_state` here
https://github.com/scipy/scipy/blob/master/scipy/_lib/_util.py#L73